vm/systemSmartContracts/governance.go

```diff
@@ -1266,6 +1330,37 @@ func (g *governanceContract) convertV2Config(config config.GovernanceSystemSCCon
 	}, nil
 }
 
+func (g *governanceContract) processLastEndedNonce() {
```

Comment:
how about this code reorg?
```GO
func (g *governanceContract) processLastEndedNonce() {
	lastEndedNonce := g.getLastEndedNonce()
	currentEpoch := g.eei.BlockChainHook().CurrentEpoch()

	maxNoncesToCheck := uint64(1000)
	nonceFound := big.NewInt(int64(lastEndedNonce))
	for nonce := lastEndedNonce + 1; nonce < lastEndedNonce+maxNoncesToCheck; nonce++ {
		nonceBig := big.NewInt(int64(nonce))
		proposal, err := g.getProposalFromNonce(nonceBig)
		if err != nil || proposal.EndVoteEpoch >= uint64(currentEpoch) {
			nonceFound = nonceBig.Sub(nonceBig, big.NewInt(1))
			break
		}
	}

	g.eei.SetStorage([]byte(lastEndedNonceKey), nonceFound.Bytes())
}
```

Reply:
in this case we have a `maxNoncesToCheck` that will stop the get operations if something goes wrong

Reply:
👍